src/llms/llm_loader.py

```diff
@@ -0,0 +1,233 @@
+# workaround to disable UserWarning
+import warnings
+warnings.simplefilter("ignore", UserWarning)
+
+from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler
+from langchain.callbacks.manager import CallbackManager
+
+import os, inspect
+
+from utils.logger import Logger
+
+class LLMLoader:
+    """
+    Note: This class loads the LLM backend libraries if the specific LLM is loaded.
+    Known caveats: Currently supports a single instance/model per backend
+
+    The llm_backends parameter is an array with the backend llm and (optional) parameters to override
+
+    # format for dictionary
+    llm_backend = {
+        'backend': 'openai',
+        'params': {...}     # optional key
+    }
+
+    # overriding specific parameters
+    llm_backend = {
+        'backend' : 'ollama',
+        'params': {'temperature': 0.02, 'top_p': 0.95},
+    }
+
+    # using default parameters
+    llm_backend = {'backend': 'tgi'}
+
+    # Using the class
+    llm_config = LLMLoader(llm_backend=llm_backend)
+    llm_chain = LLMChain(llm=llm_config.llm, prompt=prompt)
+
+    """
+    def __init__(self, llm_backend={}, logger=None) -> None:
+        self.logger = logger if logger is not None else Logger("llm_loader").logger
+        # LLM backend to use or fallback to default one
+        self.llm_backend = llm_backend.get('backend') if llm_backend.get('backend') is not None else os.environ.get('LLM_DEFAULT', {'backend':'ollama'})
```

Comment:
Consider this setting.

```python
def __init__(self, llm_backend=None, ...):
        self.llm_backend = llm_backend or os.environ.get('LLM_DEFAULT', 'ollama')
        ....
```

If `llm_backend` is provided to constructor, its used. Otherwise its picked from env (and if env is not set, ollama is used as default). You also make a clear dependency injection available in tests by this.

Reply:
ollama does not have additional dependencies (it is built in into langchain), all the other do have external libraries. Would you prefer none?

Reply:
You mean `os.environ.get('LLM_DEFAULT', None)`?

Reply:
I'm okay with fallback to None if that is the preference.

Reply:
Updated